### PR TITLE
fix(curriculum): random background random color method placement in step 3

### LINKED ISSUE #55312: Random Background Color Changer Step 3 placement of method
### Describe the Issue

A camper had an issue with code not passing due to the placement of the Math.floor() static method.

### Affected Page

https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures-v8/learn-basic-debugging-by-building-a-random-background-color-changer/step-3

### Your code

```
function getRandomIndex() {
   console.log(Math.floor(Math.random() * darkColorsArr.length));
}
```

### Expected behavior

Both
```
   console.log(Math.floor(Math.random() * darkColorsArr.length));
```
and
```
   console.log(Math.floor(darkColorsArr.length * Math.random()));
```
should pass the challenge.

### Screenshots

_No response_

### System

- Device: [e.g. iPhone 6, Laptop]
- OS: [e.g. iOS 14, Windows 10, Ubuntu 20.04]
- Browser: [e.g. Chrome, Safari]
- Version: [e.g. 22]

### Additional context

Here is a link to the forum post.
https://forum.freecodecamp.org/t/learn-basic-debugging-by-building-a-random-background-color-changer-step-3/694680/2

### PR DESCRIPTION
**Pull request by gagan-bhullar-tech**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #55312

### CODE REVIEW

**Review by majestic-owl448 — CHANGES_REQUESTED** [review]
Please make so that `console.log(Math.floor(darkColorsArr.length * darkColorsArr.length))` and `console.log(Math.floor(Math.random() * Math.random()))` are not accepted

image

**Review by lasjorg — CHANGES_REQUESTED** [review]

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-debugging-by-building-a-random-background-color-changer/663250b42513ef5975599c49.md [hunk 1]
```diff
@@ -24,7 +24,7 @@ Remember that you worked with a method in the Role Playing Game that rounds a nu
 You should round `darkColorsArr.length * Math.random()` down to the nearest whole number.
 
 ```js
-assert.match(getRandomIndex.toString(), /console\.log\(\s*Math\.floor\(\s*darkColorsArr\.length\s*\*\s*Math\.random\(\s*\)\s*\)\s*\)/);
+assert.match(getRandomIndex.toString(), /console\.log\(\s*Math\.floor\(\s*(darkColorsArr\.length\s*\*\s*Math\.random\(\s*\)|(Math\.random\(\s*\)\s*\*\s*darkColorsArr\.length))\s*\)\s*\)/);
```
Comment (lasjorg) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-debugging-by-building-a-random-background-color-changer/663250b42513ef5975599c49.md:27]:
```suggestion
assert.match(getRandomIndex.toString(), /console\.log\(\s*Math\.floor\(\s*(darkColorsArr\.length\s*\*\s*Math\.random\(\s*\)|Math\.random\(\s*\)\s*\*\s*darkColorsArr\.length)\s*\)\s*\)/);
```

I don't think we need this second capture group.

**Review by lasjorg — CHANGES_REQUESTED** [review]

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-debugging-by-building-a-random-background-color-changer/663250b42513ef5975599c49.md (hunk 1 above)
Comment (gagan-bhullar-tech) [reply curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-debugging-by-building-a-random-background-color-changer/663250b42513ef5975599c49.md]: yes. thanks for pointing out

**Review by gikf — APPROVED** [review]

**Review by majestic-owl448 — APPROVED** [review]
LGTM 👍

**Review by lasjorg — APPROVED** [review]

**Pull request merged**